addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -64,6 +67,25 @@ const Wysiwyg = Widget.extend({
         allowCommandVideo: true,
     },
     init: function (parent, options) {
+        // Redefine the getRangeAt function in order to avoid an error appearing
+        // when clicking on form fields in editable mode on Firefox.
+        // The bug comes from the fact that sometimes, the range returned by
+        // getRangeAt is "restricted" for no reason.
+        // Ex: Range { commonAncestorContainer: Restricted, startContainer: Restricted,
+        // startOffset: 0, endContainer: Restricted, endOffset: 0, collapsed: true }
+        // The proposed solution consists in detecting when the range is restricted
+        // and then redefining it manually based on the current selection.
+        Selection.prototype.getRangeAt = function (index) {
+            let range = originalGetRangeAt.call(this, index);
+            // Check if the range is restricted.
+            if (!!range.startContainer && !Object.getPrototypeOf(range.startContainer)) {
```

Comment:
`!!range.startContainer` -> useless to convert it to a boolean in a `if`, you can just write `range.startContainer`